staging/src/k8s.io/client-go/transport/transport.go

```diff
@@ -167,3 +169,15 @@ func rootCertPool(caData []byte) *x509.CertPool {
 	certPool.AppendCertsFromPEM(caData)
 	return certPool
 }
+
+func tryCancelRequest(rt http.RoundTripper, req *http.Request) {
+	type canceler interface {
+		CancelRequest(*http.Request)
+	}
+	switch rt := rt.(type) {
+	case canceler:
+		rt.CancelRequest(req)
+	case utilnet.RoundTripperWrapper:
+		tryCancelRequest(rt.WrappedRoundTripper(), req)
```

Comment:
What implements WrappedRoundTripper?

Reply:
Many transports in the k8s codebase that do not implement cancel:

```
staging/src/k8s.io/apimachinery/pkg/util/proxy/upgradeaware.go
121:// WrappedRoundTripper returns the round tripper that a caller would use.
122:func (rt *upgradeRequestRoundTripper) WrappedRoundTripper() http.RoundTripper {
455:func (rt *corsRemovingTransport) WrappedRoundTripper() http.RoundTripper {

staging/src/k8s.io/apimachinery/pkg/util/proxy/transport.go
128:func (rt *Transport) WrappedRoundTripper() http.RoundTripper {

staging/src/k8s.io/apimachinery/pkg/util/net/http.go
121:    WrappedRoundTripper() http.RoundTripper
146:            return DialerFor(transport.WrappedRoundTripper())
167:            return TLSClientConfig(transport.WrappedRoundTripper())

staging/src/k8s.io/client-go/transport/round_trippers.go
150:func (rt *authProxyRoundTripper) WrappedRoundTripper() http.RoundTripper { return rt.rt }
178:func (rt *userAgentRoundTripper) WrappedRoundTripper() http.RoundTripper { return rt.rt }
209:func (rt *basicAuthRoundTripper) WrappedRoundTripper() http.RoundTripper { return rt.rt }
269:func (rt *impersonatingRoundTripper) WrappedRoundTripper() http.RoundTripper { return rt.delegate }
328:func (rt *bearerAuthRoundTripper) WrappedRoundTripper() http.RoundTripper { return rt.rt }
455:func (rt *debuggingRoundTripper) WrappedRoundTripper() http.RoundTripper {

staging/src/k8s.io/client-go/discovery/round_tripper.go
62:func (rt *cacheRoundTripper) WrappedRoundTripper() http.RoundTripper { return rt.rt.Transport }

staging/src/k8s.io/client-go/plugin/pkg/client/auth/azure/azure.go
144:func (r *azureRoundTripper) WrappedRoundTripper() http.RoundTripper { return r.roundTripper }

staging/src/k8s.io/client-go/plugin/pkg/client/auth/openstack/openstack.go
135:func (t *tokenRoundTripper) WrappedRoundTripper() http.RoundTripper { return t.RoundTripper }

staging/src/k8s.io/client-go/plugin/pkg/client/auth/gcp/gcp.go
383:func (t *conditionalTransport) WrappedRoundTripper() http.RoundTripper { return t.oauthTransport.Base }

staging/src/k8s.io/client-go/plugin/pkg/client/auth/oidc/oidc.go
218:func (t *roundTripper) WrappedRoundTripper() http.RoundTripper { return t.wrapped }
```

Reply:
Ack, then in your test use something that implements WrappedRoundTripper and not CancelRequest (instead of oauth2.Transport)

Reply:
Done.